worlds/sc2/ItemGroups.py

```diff
@@ -77,24 +76,486 @@
 ]
-item_name_groups["Nerazim"] = [
+item_name_groups[ItemGroupNames.NERAZIM_UNITS] = [
     ItemNames.CENTURION, ItemNames.STALKER, ItemNames.DARK_TEMPLAR, ItemNames.SIGNIFIER, ItemNames.DARK_ARCHON,
     ItemNames.ANNIHILATOR,
     ItemNames.CORSAIR, ItemNames.ORACLE, ItemNames.VOID_RAY,
 ]
-item_name_groups["Tal'Darim"] = [
+item_name_groups[ItemGroupNames.TAL_DARIM_UNITS] = [
     ItemNames.SUPPLICANT, ItemNames.SLAYER, ItemNames.HAVOC, ItemNames.BLOOD_HUNTER, ItemNames.ASCENDANT,
     ItemNames.VANGUARD, ItemNames.WRATHWALKER,
     ItemNames.DESTROYER, ItemNames.MOTHERSHIP,
-    ItemNames.WARP_PRISM_PHASE_BLASTER,
 ]
-item_name_groups["Purifier"] = [
+item_name_groups[ItemGroupNames.PURIFIER_UNITS] = [
     ItemNames.SENTINEL, ItemNames.ADEPT, ItemNames.INSTIGATOR, ItemNames.ENERGIZER,
     ItemNames.COLOSSUS, ItemNames.DISRUPTOR,
     ItemNames.MIRAGE, ItemNames.TEMPEST,
-]
\ No newline at end of file
+]
+item_name_groups[ItemGroupNames.SOA_ITEMS] = soa_items = [
+    *[item_name for item_name, item_data in Items.item_table.items() if item_data.type == Items.ProtossItemType.Spear_Of_Adun],
+    ItemNames.SOA_PROGRESSIVE_PROXY_PYLON,
+]
+lotv_soa_items = [item_name for item_name in soa_items if item_name != ItemNames.SOA_PYLON_OVERCHARGE]
+item_name_groups[ItemGroupNames.PROTOSS_GLOBAL_UPGRADES] = [
+    item_name for item_name, item_data in Items.item_table.items() if item_data.type == Items.ProtossItemType.Solarite_Core
+]
+item_name_groups[ItemGroupNames.LOTV_GLOBAL_UPGRADES] = lotv_global_upgrades = [
+    ItemNames.NEXUS_OVERCHARGE,
+    ItemNames.ORBITAL_ASSIMILATORS,
+    ItemNames.WARP_HARMONIZATION,
+    ItemNames.MATRIX_OVERLOAD,
+    ItemNames.GUARDIAN_SHELL,
+    ItemNames.RECONSTRUCTION_BEAM,
+]
+item_name_groups[ItemGroupNames.LOTV_ITEMS] = vanilla_lotv_items = (
+    lotv_units
+    + protoss_buildings
+    + lotv_soa_items
+    + lotv_global_upgrades
+    + protoss_generic_upgrades
+)
+
+item_name_groups[ItemGroupNames.VANILLA_ITEMS] = vanilla_items = (
+    vanilla_wol_items + vanilla_hots_items + vanilla_lotv_items
+)
+
+
+# Sanity checks
+def _sanity_check() -> None:
+    all_progressive_upgrades = [
+        item_name for item_name, item_data in Items.item_table.items()
+        if item_data.type.display_name == Items.TerranItemType.Progressive.display_name
+    ]
+    assert len(terran_units) == len(barracks_units) + len(factory_units) + len(starport_units) + len(terran_mercenaries)
+    assert len(protoss_units) == len(gateway_units) + len(robo_units) + len(stargate_units)
+    for item_name in terran_original_progressive_upgrades:
+        assert item_name in all_progressive_upgrades
+        assert item_name in wol_upgrades
+    for item_name in terran_stimpacks:
+        assert "Stimpack" in item_name
+    for var_name, display_name in ItemGroupNames.__dict__.items():
+        if var_name.startswith("_"):
+            continue
+        assert display_name in item_name_groups
+_sanity_check()
```

Comment:
Shouldn't be a unit test instead?

Reply:
Yeah this was originally only two asserts, but things tend to grow. I'll throw together a unit test file.